doc/source/development/rfc/rfc103_schema_open_option.rst

```diff
@@ -0,0 +1,113 @@
+.. _rfc-103:
+
+===================================================================
+RFC 103 add a SCHEMA open option to selected OGR drivers
+===================================================================
+
+=============== =============================================
+Author:         Alessandro Pasotti
+Contact:        elpaso at itopen.it
+Status:         Draft
+Created:        2024-10-22
+=============== =============================================
+
+Summary
+-------
+
+This RFC enables users to specify a SCHEMA open option in the OGR
+drivers that support it.
+
+The new option will be used to override the auto-detected fields types.
+
+Motivation
+----------
+
+Several OGR drivers must guess the attribute data type: CSV, GeoJSON, SQLite,
+the auto-detected types are not always correct and the user has no way to
+override them at opening tim.
+
+For the details please see the discussion attached to the issue: https://github.com/OSGeo/gdal/issues/10943
+
+Implementation
+--------------
+
+A new open option named SCHEMA will be added to the following drivers:
+
+- CSV
+- GeoJSON
+- SQLite
+- GML
+
+The option will be used to specify the schema in the form of a JSON document (or a path to a JSON file).
+
+The schema document will not need to to contain a definition for every field and it will allow partial overrides of the auto-detected fields types.
+
+The structure of the JSON document will be similar to the one produced by the `ogrinfo -json` command
+with the notable exception that (for the scope of this RFC) only the type of the fields will be considered:
+
+.. code-block:: json
+
+    {
+    "fields": [
+        {
+        "name": "field1",
+        "type": "string"
```

Comment:
We would need also to support "subtype" (for String JSON, or Integer Boolean).  And perhaps "width" and "precision" too

Reply:
It makes sense.
Is it `subtype` also exposed by `ogrinfo -json` ?

Reply:
For `schema_type` I was thinking to make it a layer-level property, this way a combination of patch and full can be used for individual layers.

I'm working on a json schema document to clarify this.